WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostListAdapter.kt

```diff
@@ -11,30 +11,37 @@ import android.view.ViewGroup
 import org.wordpress.android.R
 import org.wordpress.android.fluxc.model.LocalOrRemoteId.LocalId
 import org.wordpress.android.fluxc.model.LocalOrRemoteId.RemoteId
+import org.wordpress.android.ui.posts.PostListItemCompactViewHolder
 import org.wordpress.android.ui.posts.PostListItemViewHolder
 import org.wordpress.android.ui.posts.PostViewHolderConfig
+import org.wordpress.android.ui.posts.ViewLayoutType
+import org.wordpress.android.ui.posts.ViewLayoutType.COMPACT
 import org.wordpress.android.ui.utils.UiHelpers
 import org.wordpress.android.viewmodel.posts.PostListItemType
 import org.wordpress.android.viewmodel.posts.PostListItemType.EndListIndicatorItem
 import org.wordpress.android.viewmodel.posts.PostListItemType.LoadingItem
 import org.wordpress.android.viewmodel.posts.PostListItemType.PostListItemUiState
 
 private const val VIEW_TYPE_POST = 0
-private const val VIEW_TYPE_ENDLIST_INDICATOR = 1
-private const val VIEW_TYPE_LOADING = 2
+private const val VIEW_TYPE_POST_COMPACT = 1
+private const val VIEW_TYPE_ENDLIST_INDICATOR = 2
+private const val VIEW_TYPE_LOADING = 3
 
 class PostListAdapter(
     context: Context,
     private val postViewHolderConfig: PostViewHolderConfig,
-    private val uiHelpers: UiHelpers
+    private val uiHelpers: UiHelpers,
+    private var viewLayoutType: ViewLayoutType
```

Comment:
This approach works as expected, however, I believe the `PostListItemUiState` doesn't match the UI now. 

For example when the list item has 5 actions available the `actions` field contains the following structure:
```
(SingleItem,SingleItem, MoreItem(SingleItem,SingleItem,SingleItem))
```
The UI presents it as
```  
(MoreItem(SingleItem,SingleItem,SingleItem,SingleItem,SingleItem))
```

What do you think about introducing new `PostListItemType` for the compact view. We wouldn't need to pass the `ViewLayoutType` in the constructor and the view model (more specifically `PostListItemUiStateHelper`) would take care of creating the correct state. I believe it'd be more consistent with the other view holder types `LoadingItem and EndListIndicatorItem`. 

Having said that, maybe it'd be too complicate and it's not worth it. However, I think we should at least try to make the `PostListItemUiState` match the UI. Wdyt? Do you have any better ideas how to go about it?

Reply:
That is true. I think you're right really - there want to be different view objects depending on the layout. It felt when I wrote it that shouldn't really be the case, as effectively we are displaying the same information but in a different shape, so really it's entirely a view decision and not a view model (as in theory, the model hasn't changed) but you're right that the menu broke that at the last minute.
I guess there are two options: 
1) Use a totally different model for the compact rows or 
2) Include a key in there for compactActions as well as actions. 
I actually don't mind doing either of these things. What's your preference?

Reply:
I usually prefer to have more structure than less, so if it were up to me I'd go with "1". The two layouts share most of the data, but not all of them (eg. the compact layout doesn't have excerpt). 

I think I'd need to see both version to be sure whether it feels right. So it's up to you, feel free to go with the easier solution if it feels like the right one.